crates/burn-jit/src/ops/base.rs

```diff
@@ -80,6 +82,54 @@ pub(crate) fn permute<R: Runtime, E: JitElement, const D: usize>(
 
     tensor
 }
+pub(crate) fn expand<R: Runtime, E: JitElement, const D: usize, const D_OUT: usize>(
+    tensor: JitTensor<R, E, D>,
+    target_shape: Shape<D_OUT>,
+) -> JitTensor<R, E, D_OUT> {
+    // Initialize new strides with zeros
+    let mut new_strides = [0usize; D_OUT];
+
+    // Calculate the difference in dimensions
+    let dim_diff = D_OUT.saturating_sub(D);
+
+    // Compare dimensions from the end, setting strides for matching dimensions or broadcasted ones
+    let mut tensor_dim_iter = tensor.shape.dims.iter().rev();
+    for i in (0..D_OUT).rev() {
+        if i >= dim_diff {
+            if let Some(&tensor_dim) = tensor_dim_iter.next() {
+                if tensor_dim == target_shape.dims[i] || tensor_dim == 1 {
+                    // Copy stride for non-broadcast dimensions or set to 0 for broadcast ones
+                    new_strides[i] = if tensor_dim == target_shape.dims[i] {
+                        tensor.strides[i - dim_diff]
+                    } else {
+                        0
+                    };
+                } else {
+                    // Error handling: Dimension mismatch for broadcasting
+                    panic!(
+                        "Dimension mismatch: cannot broadcast dimension {} of tensor to target shape",
+                        tensor_dim
+                    );
+                }
+            } else {
+                // If the input tensor has fewer dimensions, treat missing dimensions as 1
```

Comment:
Wrong comment: `treat missing dimensions as 1` => `treat missing dimensions as 0`

Reply:
It was partially correct. The stride should be set to zero but treat the missing dim as 1. I added the following:

```rust
  // If the input tensor has fewer dimensions, treat missing dimensions as 1
  // and set stride to 0 (broadcasting)
  new_strides[i] = 0;
```